samtranslator/model/api/api_generator.py

```diff
@@ -1168,7 +1169,7 @@ def _construct_authorizer_lambda_permission(self) -> List[LambdaPermission]:
 
         for authorizer_name, authorizer in authorizers.items():
             # Construct permissions for Lambda Authorizers only
-            if not authorizer.function_arn:
+            if not authorizer.function_arn or authorizer.disable_function_default_permissions:
```

Comment:
whenever I see a boolean check, I wonder are we raise exception whenever it is an intrinsic 🤔

Reply:
It should raise an exception if its not boolean, I added an error check and error test to make sure the value is boolean. If they use an intrinsic then adding Language Extensions should fix the exception if it pops up

Reply:
Oh just saw `tests/translator/input/error_api_invalid_auth.yaml`.